Doc/whatsnew/3.10.rst

```diff
@@ -987,10 +991,18 @@ inspect
 When a module does not define ``__loader__``, fall back to ``__spec__.loader``.
 (Contributed by Brett Cannon in :issue:`42133`.)
 
-Added *globalns* and *localns* parameters in :func:`~inspect.signature` and
-:meth:`inspect.Signature.from_callable` to retrieve the annotations in given
-local and global namespaces.
-(Contributed by Batuhan Taskaya in :issue:`41960`.)
+Added :func:`inspect.get_annotations`, which safely computes the annotations
+defined on an object. It works around the quirks of accessing the annotations
+on various types of objects, and makes very few assumptions about the object
+it examines. :func:`inspect.get_annotations` can also correctly un-stringize
+stringized annotations.  :func:`inspect.get_annotations` is now considered
+best practice for accessing the annotations dict defined on any Python object.
+Relatedly, :func:`inspect.signature`,
+:func:`inspect.from_callable`, and `inspect.from_function` now call
```

Comment:
Tests are failing due to this line:
```suggestion
:func:`inspect.from_callable`, and :func:`inspect.from_function` now call
```
alternatively you can just use double backticks.

Reply:
I'm a big fan of `:func:` these days, definitely going that way with it.

Reply:
And, maybe I've been thinking about this topic way too long, but I don't think the difference is subtle ;-)

Think of it this way: an annotation is defined by a syntactic context, a type hint is defined as something inheriting from `TypeVar` (or its stringized equivalent, etc etc).  It's true that lots of people only ever use type hints as annotations, and only ever annotate with type hints.  But they're still very different concepts--confusing the two is like confusing a function argument and an int.

Reply:
Yeah I agree with what you said - type hints need not be annotations, it just needs to bear some meaning to the type checker (eg. type comments). Whilst annotations is a purely syntactical construct (well bar some exceptions like in ``dataclasses`` and runtime checking)

I only noticed the major difference *while* I was writing the comment and reflecting for the 10 seconds after posting 😉 . Hence why I promptly deleted it after feeling pretty dumb hehe.

PS. regarding `

> a type hint is defined as something inheriting from TypeVar

Where can I read more about this? I didn't know and I can't find much info about that.

Reply:
Actually I misspoke.  There are a couple other things that can be type hints.  Types, for example :D

I'm not an expert on `typing` or type hints.  All I can really tell you is that `typing._type_check()` appears to be as canonical a source as one is likely to find.  It smiles upon instances of `type`,  `Any`, `NoReturn`,  `TypeVar`, `ForwardRef`, `Union`, `ParamSpec`, and some instances of `_GenericAlias`.  It complains about instances of `_SpecialForm`, the objects `Generic` and `Protocol`, or any callable not previously explicitly permitted.  And anything I didn't mention above seems to get a pass, too.